### PR TITLE
RI: Introduce "Welcome to Reader" card

### PR DESCRIPTION
**Pull request by ashiagr**
Parent Issue: #12028

This PR adds
- "Welcome to Reader" card 
- Item decorator

to new discover feed.

"Welcome to Reader" card is hidden when list is refreshed.

 Phone Portrait Light |  Phone Phone Dark
---------------| -----------------
phone_light_portrait | phone_portrait

Phone Landscape Light | Phone Landscape Dark
---------------| -----------------
 device-2020-08-20-144404| device-2020-08-20-144444

Tablet Portrait Light | Tablet Landscape Light
---------------| -----------------
tablet_portrait_light| tablet_landscape_light

To test:
Prerequisite: RI FF is enabled

1. Clean app data
2. Launch app, login and go to Reader -> Discover tab
3. Notice that "Welcome Banner" card is shown
4. Pull to refresh
5. Notice that  "Welcome Banner" card is gone

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/res/values/colors.xml
Comment (ashiagr) [inline WordPress/src/main/res/values/colors.xml:331]: Had to add these colors in `colors.xml` as I couldn't find any existing palette color that could be used on both light and dark modes for welcome card text and background.

File: WordPress/src/main/res/drawable/bg_colorful_dots.xml
Comment (zwarm) [inline WordPress/src/main/res/drawable/bg_colorful_dots.xml]: @ashiagr wdyt about replacing with color resource references instead of hex values?

File: WordPress/src/main/res/drawable/bg_colorful_dots_large.xml
Comment (zwarm) [inline WordPress/src/main/res/drawable/bg_colorful_dots_large.xml]: @ashiagr same as with bg_colorful_dots.xml? wdyt?

File: WordPress/src/main/res/values/colors.xml
Comment (zwarm) [reply WordPress/src/main/res/values/colors.xml:331]: iirc we had to change the empty slate view to match colors within the palette. I wonder if we need to adjust here as well, but that's more of a question for @mbshakti

**Review by zwarm — COMMENTED** [review]
Nice work! I left a couple of comments related to colors. Let me know what you think.

File: WordPress/src/main/res/drawable/bg_colorful_dots_large.xml
Comment (ashiagr) [reply WordPress/src/main/res/drawable/bg_colorful_dots_large.xml]: Replaced here: 463e54e
Couldn't find colors for red and blue dots. Will check with Shakti and update.

File: WordPress/src/main/res/drawable/bg_colorful_dots.xml
Comment (ashiagr) [reply WordPress/src/main/res/drawable/bg_colorful_dots.xml]: Replaced here: 463e54e

File: WordPress/src/main/res/values/colors.xml
Comment (ashiagr) [reply WordPress/src/main/res/values/colors.xml:331]: Will check with Shakti and update.

Comment (malinajirka) [reply WordPress/src/main/res/values/colors.xml:331]: Yep, I agree that we might want to update the colors in the design instead of introducing a new set of colors.

File: WordPress/src/main/res/values/reader_styles.xml
Comment (malinajirka) [inline WordPress/src/main/res/values/reader_styles.xml]: I'm wondering if `text` can be considered as a style. I'd personally move it to the layout file or perhaps even the VM. Wdyt

File: WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt]: I feel like the card is not related to the server request at all and the ReaderDiscoverLogic is a class which responsibility is to sync the local db with the server db. Could we perhaps set the flag [here](https://github.com/wordpress-mobile/WordPress-Android/blob/26bc872cdd7f8ca277ab9ef7209ad508c1bbe4a2/WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModel.kt#L166)? Wdyt?

**Review by malinajirka — COMMENTED** [review]
Thanks @ashiagr! The changes look great. I've left a couple of questions/suggestions. Let me know what you think ;).

File: WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt
Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt]: Ohh, the getDiscoverCardsUseCase is invoked multiple times if you loadMore items. I can see the issue. Perhaps, we could set it in viewModel.onCleared?

File: WordPress/src/main/res/values/reader_styles.xml
Comment (ashiagr) [reply WordPress/src/main/res/values/reader_styles.xml]: Since it was a static text I included it within styles as I saw this convention at another place in the code [here](https://github.com/wordpress-mobile/WordPress-Android/blob/315c5ea072eba1d00121ce898764536ffc8e654f/WordPress/src/main/res/values/styles.xml#L889). But I see your point. I'll move it to the VM.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt
Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt]: Setting it on `viewModel.onCleared` won't clear welcome card on pull to refresh. I'm thinking of setting at both `viewModel.onCleared` and `viewModel.swipeToRefresh()`.

Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt]: 👍 Sounds good to me.

Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt]: Fixed in bff2372

File: WordPress/src/main/res/values/reader_styles.xml
Comment (ashiagr) [reply WordPress/src/main/res/values/reader_styles.xml]: Moved in ddf6831

File: WordPress/src/test/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModelTest.kt
Comment (malinajirka) [inline WordPress/src/test/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModelTest.kt]: I'm wondering if we should add a test to make sure we don't show the ContentUiState when the only available data is the WelcomeCard -> in other words, that we don't show the welcome card when the discover feed is empty. Wdyt?

**Comment by malinajirka** [comment]
Thanks @ashiagr! The changes LGTM - I've left one minor comment regarding the unit test, but I wouldn't block this PR on it. However, the colors.xml issue is still open - do you want to merge this PR and resolve it later or are you planning on resolving it here?

**Comment by ashiagr** [comment]
> do you want to merge this PR and resolve it later or are you planning on resolving it here

I plan to resolve it here, plus I've to replace svg files with the new ones that Shakti provided.
Moving this PR to draft till then.

File: WordPress/src/main/res/drawable/bg_colorful_dots_large.xml
Comment (ashiagr) [reply WordPress/src/main/res/drawable/bg_colorful_dots_large.xml]: Replaced red and blue dot colors that I got from Shakti in ccc1f86

**Comment by ashiagr** [comment]
Replaced welcome card svg in 340b2ca, new screenshots: 
(approved by @mbshakti)

phone portrait | phone landscape
---------------|-----------------
welcome_phone_por | welcome_phone_land

tablet portrait | tablet landscape
---------------|-----------------
welcome_tab_por | welcome_tab_land

File: WordPress/src/test/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModelTest.kt
Comment (ashiagr) [reply WordPress/src/test/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModelTest.kt]: Test added in 58a61ca

File: WordPress/src/main/res/values/colors.xml
Comment (ashiagr) [reply WordPress/src/main/res/values/colors.xml:331]: I discussed with Shakti, quoting her:

>  it’s such a custom/rare card that we probably won’t reuse else where

With this limitation, I think we can keep these colors here (note that they're using existing colors but not from the light or dark style color attributes). 

We can also introduce new style color attributes in the `values/style.xml` and `values-night/style.xml` similar to `colorGutenbergBackground`:
```
<item name="colorReaderWelcomeCardBackground">@color/primary_emphasis_lowest</item>
<item name="colorReaderWelcomeCardText">@color/blue_70</item>
```

Let me know what you think.

Also @zwarm regarding this:
> we had to change the empty slate view to match colors within the palette

Are you talking about `reader_following_empty_view_background` color? I see it is added to `values/colors.xml`,  `values-night/colors.xml` and not picked from style color attributes in `values/style.xml` and `values-night/style.xml`.

File: WordPress/src/test/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModelTest.kt
Comment (malinajirka) [reply WordPress/src/test/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModelTest.kt]: Awesome, thanks!

File: WordPress/src/main/res/values/colors.xml
Comment (malinajirka) [reply WordPress/src/main/res/values/colors.xml:331]: If I understand our approach for the colors.xml correctly this addition is still not following it.

I believe it's ok-ish to add `reader_welcome_card_background` and `reader_welcome_card_text`, since these two entries need to have custom colors in light vs dark mode. However, I'm wondering if `reader_welcome_green_dot`, `reader_welcome_purple_dot`, ...., should be inlined directly into the drawable ?!?. 

P.S. Ideally, we wouldn't be using the colors(green, purple, yellow) directly, but we'd be using semantic colors - however, I believe it's fine to use the ["STUDIO 2.2.0"](https://github.com/wordpress-mobile/WordPress-Android/blob/d7146bac7f9ee4cc006491c6ed9b5b7818bf41a0/WordPress/src/main/res/values/colors.xml#L148) directly for this usecase as it's an illustration.

Comment (ashiagr) [reply WordPress/src/main/res/values/colors.xml:331]: > I believe it's fine to use the "STUDIO 2.2.0" directly for this usecase as it's an illustration.

Used them directly here: cc99e75

Comment (zwarm) [reply WordPress/src/main/res/values/colors.xml:331]: > Are you talking about reader_following_empty_view_background color? I see it is added to values/colors.xml, values-night/colors.xml and not picked from style color attributes in values/style.xml and values-night/style.xml.

Yep, for that they needed the custom colors for light/dark, but the others were changed to palette. I believe the changes you made are good to go. 👍

**Review by malinajirka — APPROVED** [review]
Thanks @ashiagr! LGTM 🚢  (feel free to merge it when the conflict is resolved)

**Pull request merged**
